src/java.base/share/classes/java/lang/String.java

```diff
@@ -2156,6 +2156,10 @@ public boolean regionMatches(int toffset, String other, int ooffset, int len) {
              (ooffset > (long)other.length() - len)) {
             return false;
         }
+        // Any strings match if length < 0
```

Comment:
I don't think this fix works, because the specification says:

*The result is false if and only if at least one of the following is true:*
- `toffset` is negative.
- `ooffset` is negative.
- `toffset+len` is greater than the length of this String object.
- `ooffset+len` is greater than the length of the other argument.
- There is some nonnegative integer `k` less than `len` such that: `this.charAt(toffset + k) != other.charAt(ooffset + k)`

Now consider `"12345".regionMatches(10, "012345", 1, -3)`

With the current implementation, the call returns `false` because `toffset + len` == `10 + (-3)` == 7 > 5 == `"12345".length()`. With your fix, the call will succeed because `len` == `-3` < `0`.